modules/two_factor_authentication/spec/features/my_two_factor_devices_spec.rb

```diff
@@ -68,14 +68,16 @@
     expect(page).to have_css(".on-off-status.-enabled")
 
     # Create another one as totp
-    # Visit create button
     visit my_2fa_devices_path
-    find(".toolbar-item .button", text: "2FA device").click
-    expect(page).to have_css("h2", text: I18n.t("two_factor_authentication.devices.add_new"))
-    expect(page).to have_current_path new_my_2fa_device_path, ignore_query: true
-
+    menu_button = find("[data-test-selector='two_factor_authentication_devices'] button")
+    menu_button.click
+    wait_for_network_idle if using_cuprite?
+    expect(page).to have_css("[data-test-selector='two_factor_authentication_devices_totp']")
     # Select totp
-    find(".mobile-otp-new-device-totp .button").click
+    totp_menu_item = find("[data-test-selector='two_factor_authentication_devices_totp']")
+    totp_menu_item.click
+    expect(page).to have_css("h1", text: I18n.t("two_factor_authentication.devices.add_new"))
+    expect(page).to have_current_path new_my_2fa_device_path, ignore_query: true
```

Comment:
See my comments above. Please use the test_selector helper methods instead of constructing the selector by yourself.